pycognito/__init__.py

```diff
@@ -889,3 +895,106 @@ def admin_update_user_pool_client(self, pool_id: str, client_id: str, **kwargs):
             ClientId=client_id,
             **kwargs,
         )
+
+    def associate_software_token(self):
+        """
+        Get the SecretCode used for Software Token. SecretCode use to set up Software Token MFA.
+        :return: SecretCode
+        :rtype: string
+        """
+        response = self.client.associate_software_token(AccessToken=self.access_token)
+        return response["SecretCode"]
+
+    def verify_software_token(self, code, device_name):
+        """
+        Verify the value generated by TOTP to complete the registration of Software Token MFA.
+        :param code: The value generated by TOTP
+        :param device_name: Device name to register (optional)
+        :return: verify success
+        :rtype: bool
+        """
+        response = self.client.verify_software_token(
+            AccessToken=self.access_token, UserCode=code, FriendlyDeviceName=device_name
+        )
+        return response["Status"] == "SUCCESS"
+
+    def set_user_mfa_preference(self, sms_mfa, software_token_mfa, preferred=None):
+        """
+        Register the preference of MFA.
+        :param sms_mfa: Enable SMS MFA.
+        :type sms_mfa: bool
+        :param software_token_mfa: Enable Software Token MFA.
+        :type software_token_mfa: bool
+        :param preferred: Which is the priority, SMS or Software Token? The expected value is "SMS" or "SOFTWARE_TOKEN". However, it is not needed only if both of the previous arguments are False.
+        :type preferred: string
+        :return:
+        """
+        sms_mfa_settings = {"Enabled": bool(sms_mfa), "PreferredMfa": False}
+        software_token_mfa_settings = {
+            "Enabled": bool(software_token_mfa),
+            "PreferredMfa": False,
+        }
+        if not (bool(sms_mfa) or bool(software_token_mfa)):
+            # Disable MFA
+            pass
+        if preferred == "SMS":
```

Comment:
The parameter doc says "However, `preferred` is not needed only if both of the previous arguments are False." So I think this supposed to be `elif`?